### PR TITLE
libssh2: update to 1.10.0

### PR DESCRIPTION
**Pull request by herbygillot**
- use `openssl` portgroup

Fixes: https://trac.macports.org/ticket/64151

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
LGTM. Anyone else?

**Comment by fs111** [comment]
I am the reporter of https://trac.macports.org/ticket/64151 and I can confirm that this fixes the problem. I tried it locally and mc +sftp works again for me.

**Comment by mascguy** [comment]
> I am the reporter of https://trac.macports.org/ticket/64151 and I can confirm that this fixes the problem. I tried it locally and mc +sftp works again for me.

Great to hear!

We should wait for a blessing from the maintainer - or for 72 hours to pass - before merging.

In the interim, do any fellow reviewers have any thoughts/feedback on this PR?

File: devel/libssh2/Portfile
Comment (reneeotten) [inline devel/libssh2/Portfile:36]: this should likely become `depends_lib-append` as not to overwrite dependencies from the PG (e.g. `openssl`)?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: devel/libssh2/Portfile
Comment (reneeotten) [reply devel/libssh2/Portfile]: ah, well that's good! There might be some magic happening that it does that somehow after "finishing" the rest of the Portfile. Otherwise, I would think that that dependency is added when you include the PG and then the use of `depends_lib` would overwrite it... In any case, using `depends_lib-append` never hurts ;)

**Review by reneeotten — APPROVED** [review]

**Pull request merged**
